src/common/graphql.js

```diff
@@ -1,43 +1,79 @@
 import { ApolloClient, InMemoryCache, gql } from '@apollo/client'
+import { utils } from 'ethers'
 import defaults from './defaults'
 
-const client = new ApolloClient({
-	uri: defaults.api.graphUrl,
-	cache: new InMemoryCache(),
-})
-
-const getXVaderPriceByBlock = async (block) => {
-	const tokensQuery = block ? `
-		query {
-			globals(block: { number: ${block} }) {
-				id
-				name
-				value
-			}
+const getXVaderPrice = async (type = 'Hour', skip = 0, first = 1) => {
+	const query = skip > 0 ? `
+	{
+		globals(
+			first: ${first}
+			skip: ${skip},
+			orderBy: timestamp,
+			orderDirection: desc,
+			where:{ 
+				name: "XVADER_PRICE",
+				type: "${type}"
+			}) {
+			id
+			name
+			value
+			type
+			timestamp
 		}
-	` : `
-		query {
-			globals {
-				id
-				name
-				value
-			}
+	}
+	` :	`
+	{
+		global(id: "XVADER_PRICE") {
+			id
+			name
+			value
 		}
+	}
 	`
 	try {
-		const result = await client.query({ query: gql(tokensQuery) })
-		if (!result || !result.data || !result.data.globals) {
-			return null
+		const result = await queryGraphQL(query, defaults.api.graphql.uri.vaderProtocol)
+		if (result) {
+			return result
+		}
+	}
+	catch (err) {
+		console.log(err)
+	}
+}
+
+const getXVaderApr = async (type) => {
+	const [currentPrice, daysAgoPrice] = await Promise.all([
+		getXVaderPrice(),
+		getXVaderPrice(type, 3),
+	])
+	const currentPriceBN = utils.parseUnits(currentPrice.global.value, 'wei')
+	const previousPrice = utils.parseUnits(daysAgoPrice.globals?.[0].value, 'wei')
+	const apr = ((((currentPriceBN.sub(previousPrice))
+		.mul(utils.parseUnits('1', 18)))
+		.div(previousPrice))
+		.mul(365))
+		.div(7)
+		.div(24)
```

Comment:
maybe can make this more general.
Fx: type = Hour => x = 1/24
type = 1 day => x = 1
type = 7 days => x = 7
then apr = roi * 365 / x

Reply:
If you can do make PR based on Kovan and PR into Kovan

Reply:
oke

Reply:
but we'll change it later anyways

Reply:
hmm ok, if it's temporary, then just keep it like that. Only change the formula as I mention on tg : apr = roi * 24 * 365. Could u help quick change it later?

Reply:
Yes sure, you can prepare longer period like 7 days or so once there' enough data we will update.

Reply:
Yep, That i fixed already in WIP branch. Just merged.